meilisearch-http/src/extractors/authentication/mod.rs

```diff
@@ -57,7 +57,7 @@ impl<P, D> GuardedData<P, D> {
                 }),
                 None => Err(AuthenticationError::IrretrievableState.into()),
             },
-            None => Err(AuthenticationError::MissingAuthorizationHeader.into()),
```

Comment:
> Any hints as to where should the change be made?

I would say that this is your issue. `Self::authenticate` don't return `None` only because there is no master key. I suggest changing the return value of `Self::authenticate` in order to make the difference; OR the `auth: AuthController` can inform you if a master key is set, matching on it could be a solution.

Reply:
> > Any hints as to where should the change be made?
> 
> I would say that this is your issue. `Self::authenticate` don't return `None` only because there is no master key. I suggest changing the return value of `Self::authenticate` in order to make the difference; OR the `auth: AuthController` can inform you if a master key is set, matching on it could be a solution.

Thanks!. I have implemented a solution; seems a bit hackish.